pandas/io/formats/excel.py

```diff
@@ -547,28 +551,27 @@ def _format_header_mi(self):
 
         if self.merge_cells:
             # Format multi-index as a merged cells.
-            for lnum in range(len(level_lengths)):
-                name = columns.names[lnum]
-                yield ExcelCell(lnum, coloffset, name, self.header_style)
+            for lnum, name in enumerate(columns.names):
+                yield ExcelCell(
+                    row=lnum,
+                    col=coloffset,
+                    val=name,
+                    style=self.header_style,
+                )
 
             for lnum, (spans, levels, level_codes) in enumerate(
                 zip(level_lengths, columns.levels, columns.codes)
             ):
                 values = levels.take(level_codes)
                 for i in spans:
-                    if spans[i] > 1:
-                        yield ExcelCell(
-                            lnum,
-                            coloffset + i + 1,
-                            values[i],
-                            self.header_style,
-                            lnum,
-                            coloffset + i + spans[i],
-                        )
-                    else:
-                        yield ExcelCell(
-                            lnum, coloffset + i + 1, values[i], self.header_style
-                        )
+                    yield ExcelCell(
+                        row=lnum,
+                        col=coloffset + i + 1,
+                        val=values[i],
+                        style=self.header_style,
+                        mergestart=lnum if spans[i] > 1 else None,
```

Comment:
maybe `spans[i] > 1` could first be assigned to another appropriately named variable to clarify intent and (slightly) reduce duplication.

Reply:
I am a bit confused what ``spans[i] > 1`` means. I guess it is whether there is a multi-level col. So, I used the variable ``is_multilevel``. Please let me know what you think.

Reply:
I think we are always multilevel in this function. maybe `needs_merge_cells` or `spans_multiple_cells`

Reply:
These are good names indeed. Thank you!